build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoZipBallStreamer.java

```diff
@@ -349,4 +364,29 @@ private String getFirstPathElement(String path) {
         path = StringUtils.removeStart(path, "/");
         return StringUtils.substringBefore(path, "/");
     }
+
+    /**
+     * Sets the Go version based on the content of the go.mod file in the root of the zip file.
+     */
+    private void setGoVersionByModFile(ZipArchiveEntry entry) {
+        // Go version 1.13.0 is the version where we should start looking for go.mod file.
+        if (goModVersion.equals(Version.of(1, 12, 0))) {
+            try (InputStream inputStream = zipFile.getInputStream(entry)) {
+                String modFileContent = IOUtils.toString(inputStream, StandardCharsets.UTF_8.name());
+                if (!StringUtils.isEmpty(modFileContent)) {
+                    String goVersion = Arrays.stream(
+                                    modFileContent.split("\n")).filter(line -> line.startsWith("go"))
+                            .findFirst().get().replace("go", "").trim();
+                    Optional<Version> parsedVersion = Version.tryParse(goVersion);
+                    if (!parsedVersion.isPresent()) {
+                        goVersion = goVersion + ".0"; // Ensure the version has a patch.
```

Comment:
it is also possible that from GO 1.21 it can have patch version in go.mod file for example 1.22.1 is also valid make sure .0 is appended only when the go version has major and minor.

https://go.dev/ref/mod#go-mod-file